chainercv/datasets/sbd/sbd_instance_segmentation_dataset.py

```diff
@@ -0,0 +1,123 @@
+
+    Args:
+        data_dir (string): Path to the root of the training data. If this is
+            :obj:`auto`, this class will automatically download data for you
+            under :obj:`$CHAINER_DATASET_ROOT/pfnet/chainercv/sbd`.
+        split ({'train', 'val', 'trainval'}): Select a split of the dataset.
+
+    """
+
+    def __init__(self, data_dir='auto', split='train'):
+        _check_available()
+
+        if split not in ['train', 'trainval', 'val']:
+            raise ValueError(
+                'please pick split from \'train\', \'trainval\', \'val\'')
+
+        if data_dir == 'auto':
+            data_dir = sbd_utils.get_sbd()
+
+        id_list_file = os.path.join(
+            data_dir, '{}_voc2012.txt'.format(split))
+        self.ids = [id_.strip() for id_ in open(id_list_file)]
+
+        self.data_dir = data_dir
+
+    def __len__(self):
+        return len(self.ids)
+
+    def get_example(self, i):
+        """Returns the i-th example.
+
+        Returns a color image, bounding boxes, masks and labels. The color
+        image is in CHW format.
+
+        Args:
+            i (int): The index of the example.
+
+        Returns:
+            tuple of color image, bounding boxes, masks and labels whose
+            shapes are (3, H, W), (R, 4), (R, H, W), (R, ) respectively.
+            H and W are height and width of the images, and R is the number
+            of objects in the image. The dtype of the color image and
+            the bounding boxes are :obj:`numpy.float32`, that of the
+            masks are :obj: `bool`, and that of the labels are
+            :obj:`numpy.int32`.
+
+        """
+        data_id = self.ids[i]
+        img_file = os.path.join(
+            self.data_dir, 'img', data_id + '.jpg')
+        img = read_image(img_file, color=True)
+        seg_img, ins_img = self._load_seg_ins(data_id)
+        bbox, mask, label = self._prepare_data(seg_img, ins_img)
+        return img, bbox, mask, label
+
+    def _load_seg_ins(self, data_id):
+        seg_file = os.path.join(
+            self.data_dir, 'cls', data_id + '.mat')
+        ins_file = os.path.join(
+            self.data_dir, 'inst', data_id + '.mat')
+        seg_mat = scipy.io.loadmat(seg_file)
+        seg_img = seg_mat['GTcls']['Segmentation'][0][0].astype(np.int32)
+        ins_mat = scipy.io.loadmat(ins_file)
+        ins_img = ins_mat['GTinst']['Segmentation'][0][0].astype(np.int32)
+        ins_img[ins_img == 0] = -1
+        ins_img[ins_img == 255] = -1
+        return seg_img, ins_img
+
+    def _prepare_data(self, seg_img, ins_img):
+        bbox = []
+        mask = []
+        label = []
+        instances = np.unique(ins_img)
+        for inst_id in instances[instances != -1]:
+            mask_inst = ins_img == inst_id
+            instance_class = np.unique(seg_img[mask_inst])[0] - 1
+
+            assert inst_id != -1
+            assert instance_class != -1
+
+            where = np.argwhere(mask_inst)
```

Comment:
`inst`, `ins_*`, `instances` are used.
Please be consistent.

I would like to make a suggestion.

+ `instances` --> `inst_ids`
+ `ins_img` --> `inst_img` 
+ `ins_file` --> `inst_file`

Also, it is weird that `instance_class` is an element of `label`. Same goes for `mask_inst`.
How about
+ `instance_class` --> `lbl`  or just `label.append(np.unique(seg_img[mask_inst])[0] - 1)`
+ `mask_inst` --> `m`

Reply:
I prefer `inst_*`, so I will update.